src/openms/include/OpenMS/CHEMISTRY/ISOTOPEDISTRIBUTION/IsoSpecWrapper.h

```diff
@@ -0,0 +1,357 @@
+// $Maintainer: Hannes Rost $
+// $Authors: Hannes Rost, Michał Startek, Mateusz Łącki $
+// --------------------------------------------------------------------------
+
+#pragma once
+
+#include <vector>
+
+#include <OpenMS/CONCEPT/Types.h>
+#include <OpenMS/CONCEPT/Constants.h>
+#include <OpenMS/CONCEPT/Exception.h>
+
+#include <OpenMS/KERNEL/Peak1D.h>
+
+#include <OpenMS/CHEMISTRY/EmpiricalFormula.h>
+#include <OpenMS/CHEMISTRY/ISOTOPEDISTRIBUTION/IsotopeDistribution.h>
+
+
+// Override IsoSpec's use of mmap whenever it is available
+#define ISOSPEC_GOT_SYSTEM_MMAN false
+#define ISOSPEC_GOT_MMAN false
+
+#include <OpenMS/../../thirdparty/IsoSpec/IsoSpec/isoSpec++.h>
+
+
+namespace OpenMS
+{
+
+  /**
+    * @brief Interface to the IsoSpec algorithm.
+    * 
+    * Provides an interface to the IsoSpec algorithm.
+    *
+    * @code
+    * Łącki MK, Startek M, Valkenborg D, Gambin A.
+    * IsoSpec: Hyperfast Fine Structure Calculator.
+    * Anal Chem. 2017 Mar 21;89(6):3272-3277. doi: 10.1021/acs.analchem.6b01459.
+    * @endcode
+    *
+    **/
+  class OPENMS_DLLAPI IsoSpecWrapper
+  {
+  /**
+   * @brief Interface for the IsoSpec algorithm - a generator of infinitely-resolved theoretical spectra.
+   */
+
+public:
+
+    /**
+      * @brief Run the algorithm
+      *
+      * This method will run the algorithm with parameters as set up by the constructor. It will return an
+      * IsotopeDistribution containing the observed configurations. The configurations are explicitly stored
+      * in memory, which may become a problem when considering some especially large distributions. If this,
+      * or (a rather small) performance overhead is a concern, then the generator methods (below) should be
+      * used instead.
+      *
+      * This method is provided for convience. As calling that method invalidates the object (the method should
+      * not be called again, nor anythong other than destroying the object should be done with it), the most common
+      * usage pattern of IsoSpecWrapper classes with the run method is:
+      *
+      * IsotopeDistribution dist = IsoSpecWrapperSubclass(...).run();
+      * do something with dist;
+      *
+      * @note Calling this method invalidates the object! In future versions this limitation might be removed.
+      *
+      * @note This method should not be mixed with the generator methods - a given object should only ever have
+      * its run method used, or only its generator methods used.
+      *
+      **/
+    virtual IsotopeDistribution run() = 0;
+
+    /**
+     * @brief Move the generator to a next isotopologue
+     *
+     * Advance the internal generator to the next isotopologue. The value returned determines whether the
+     * generator has been exhausted (that is, all eligible configurations have already been visited).
+     * It is invalid to call any other generator methods before the first call to nextConf(), as well as
```

Comment:
hmm would then an interface like bool result = iso.getNextConf(Peak1D&) be better?
then one could write:
while(iso.getNextConf(p))
{
  do something with p
}